spec.html

```diff
@@ -125,6 +125,38 @@ <h1>Updated Productions</h1>
   </emu-clause>
 </emu-clause>
 
+  <emu-clause id="sec-coalesce-class-elements" aoid="CoalesceClassElements">
+    <h1>CoalesceClassElements ( _elements_ )</h1>
+    <emu-alg>
+      1. Let _newElements_ be an empty List.
+      1. For _element_ in _elements_,
+        1. If _element_.[[Kind]] is `"method"` and _newElements_ contains a Record _other_ where _other_.[[Kind]] is `"method"`, SameValue(_other_.[[Key]], _element_.[[Key]]) is *true*, and _other_.[[Placement]] is _element_.[[Placement]],
+          1. If _element_.[[Decorators]] is not empty,
+            1. If _other_.[[Decorators]] is not empty, throw a *ReferenceError* exception.
+            1. Set _other_.[[Decorators]] to _element_.[[Decorators]].
+          1. If _element_.[[Key]] is a Private Name,
+            1. Assert: _element_.[[Kind]] is `"method"`.
+            1. If _element_.[[Descriptor]] has a [[Get]] field,
```

Comment:
Could we use abstract operations here, like IsAccessorDescriptor/isMutatorDescriptor or something similar? I feel like this algorithm here would be more readable if all the shape assertions and branching were extracted out.

Reply:
Do you have a particular other wording that you'd like to propose?

Reply:
I'm not sure if increasing the indirection here will make this algorithm more readable. You're talking about factoring out just those two lines for Get and Set?

Reply:
My thinking is that it would at a minimum collapse lines 288-294 to "assert isDataDescriptor + CoalesceAccessorDescriptor()", and lines lines 300-302 to the same.

Reply:
Oh, I see, somehow I missed the repetition between the private and public cases. Sure, I'll make this change.